content/posts/2025-02-26-whats_new_1_8.md

```diff
@@ -0,0 +1,295 @@
+
+To enable this feature in FlowCollector, enter `oc edit flowcollector` and configure the following in the **ebpf** section:
+
+```
+spec:
+  agent:
+    ebpf:
+      sampling: 1  # recommended so all flows are observed
+      privileged: true
+      features:
+      - NetworkEvents
+```
+
+<p style="text-align: center">Listing 4: Enable NetworkEvents feature in FlowCollector</p>
+
+A packet could be dropped for many reasons, but often times, it's due to a network policy.  In OCP, there are three different types of network policies.  Network Events supports all three.  Click the links to learn how to configure them.
+
+1. [Kubernetes Network Policy](https://docs.openshift.com/container-platform/4.18/networking/network_security/network_policy/about-network-policy.html) - similar to a firewall rule, typically based on an IP address, port, and protocol
+2. [Admin Network Policy](https://docs.openshift.com/container-platform/4.18/networking/network_security/AdminNetworkPolicy/ovn-k-anp.html) - allows cluster admin to create policies that are at the cluster level
+3. [Egress Firewall](https://docs.openshift.com/container-platform/4.18/networking/network_security/egress_firewall/viewing-egress-firewall-ovn.html) - controls what external hosts a pod can connect to
+
+In Figure 3, the Traffic flows table shows a column for Network Events.  The first row shows a drop from a network policy and the second row allowed the packet in an admin network policy.
+
+![Flows table with Network Events]({page.image('whats-new-1-8/flows-network_events.png')})
+Figure 3: Flows table with Network Events
+
+### eBPF Flow Filter Enhancements
+
+eBPF flow filter was first introduced in Network Observability 1.6.  It lets you decide what you want to observe at the initial collector stage, so you can minimize the resources used by observability.  It comes with more enhancements in this release.
+
+#### More than one flow filter rule
+
+Instead of one filter rule, you can now have up to 16 rules, thus removing a major limitation of this feature.  The order of preference is based on the most specific match, which is the longest prefix match of the CIDR or if tied, the longest prefix match of the peer CIDR.  For example, if one rule has a CIDR of 10.0.0.0/16 and another with 10.0.1.0/24, then if the address is 10.0.1.128, it would match the second rule because that's more specific (24-bit prefix vs. 16-bit prefix).  This also implies no two rules can have the same CIDRs.
+
+#### Peer CIDR
+
+`peerCIDR` is a new option that specifies the address on the server side, which is the destination when you make a request and the source when it makes a response.  There still exists a `peerIP` that can only specify a host.  However, with `peerCIDR`, you can simply use it to specify a host (/32) or a subnetwork.
+
+In summary, use `cidr` for the client side address and `peerCIDR` for the server side address.
```

Comment:
You could add that it allows to distinguish internal versus external traffic, typically 

```yaml
          cidr: 10.128.0.0/14
          peerCIDR: 10.128.0.0/14
```
stands for internal traffic (pod to pod), and all the rest is external.

For instance a good use-case is if you want to sample 1 all external traffic and sample 50 all internal, you would write:

```yaml
      rules:
        - action: Accept
          cidr: 0.0.0.0/0
          sampling: 1
        - action: Accept
          cidr: 10.128.0.0/14
          peerCIDR: 10.128.0.0/14
          sampling: 50
        - action: Accept
          cidr: 172.30.0.0/16      # for services
          peerCIDR: 10.128.0.0/14
          sampling: 50
```
(this was one of the requirement use-case that drove this enhancement)

Reply:
Also:
> use `cidr` for the client side address and `peerCIDR` for the server side address.

it can be said this way for simplified guidance, but it doesn't really matter where is the client and where is the server. It could even over-complicate the reasoning if the user starts thinking about where the server sides are located, and where the clients sides are.

Reply:
I used this as the example.  I believe the last rule should be:
```
        - action: Accept
          cidr: 10.128.0.0/14
          peerCIDR: 172.30.0.0/16      # for services
          sampling: 50
```
          
I flipped `cidr` and `peerCIDR`.

Reply:
yeah, in fact both work the same, since we also flip & check cidr to check for the return traffic - that's why the client/server distinction isn't important. But your variant is probably better / easier to understand.